File: app/src/main/java/test/itexico/movies/view/EpisodesActivity.java
package test.itexico.movies.view;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.constraint.ConstraintLayout;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentTransaction;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.RecyclerView;
import android.widget.FrameLayout;

import test.itexico.movies.R;

public class EpisodesActivity extends AppCompatActivity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        setContentView(R.layout.activity_container);
        FrameLayout container = findViewById(R.id.main_container);

        Bundle extras = getIntent().getExtras();

        if (container != null) {
            if (savedInstanceState != null) {
                return;
            }
            FragmentManager fragmentManager = getSupportFragmentManager();
            FragmentTransaction fragmentTransaction = fragmentManager.beginTransaction();
            EpisodesFragment episodesFragment = new EpisodesFragment();
            episodesFragment.setArguments(extras);
            fragmentTransaction.add(R.id.main_container, episodesFragment);
            fragmentTransaction.commit();
        }
    }

}

File: app/src/main/java/test/itexico/movies/view/EpisodesActivity.kt
package test.itexico.movies.view

import android.arch.lifecycle.LifecycleOwner
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.widget.FrameLayout
import kotlinx.android.synthetic.main.activity_container.*
import test.itexico.movies.R

class EpisodesActivity : AppCompatActivity(), LifecycleOwner {

    internal var container: FrameLayout? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_container)
        container = main_container
        val extras = intent.extras
        if (container != null) {
            if (savedInstanceState != null) {
                return
            }
            val fragmentManager = supportFragmentManager
            val fragmentTransaction = fragmentManager.beginTransaction()
            val episodesFragment = EpisodesFragment()
            episodesFragment.arguments = extras
            fragmentTransaction.add(R.id.main_container, episodesFragment)
            fragmentTransaction.commit()
            lifecycle.addObserver(episodesFragment)
        }
    }

}
